cmake/CPM.cmake

```diff
@@ -230,6 +231,19 @@ function(CPMAddPackage)
 
   if (DEFINED CPM_ARGS_GIT_TAG)
     list(APPEND CPM_ARGS_UNPARSED_ARGUMENTS GIT_TAG ${CPM_ARGS_GIT_TAG})
+
+    # If GIT_SHALLOW is explicitly specified, hornor the value.
+    # Otherwise, enable shallow clone only when GIT_TAG is not a commit hash.
+    # Our guess may not be accurate, but it should guarantee no commit hash get mis-detected.
+    if (DEFINED CPM_ARGS_GIT_SHALLOW)
+      list(APPEND CPM_ARGS_UNPARSED_ARGUMENTS GIT_SHALLOW ${CPM_ARGS_GIT_SHALLOW})
+    else()
+      cpm_is_git_tag_commit_hash(${CPM_ARGS_GIT_TAG} IS_HASH)
+      if (NOT ${IS_HASH})
+        message(STATUS "GIT_TAG=${CPM_ARGS_GIT_TAG} is not a commit hash; enable shallow clone")
```

Comment:
I wouldn't print internal config decisions to avoid flooding the output.

Reply:
Honestly, that was the first time I didn't use `CPM_SOURCE_CACHE`; and it indeed caused a checkout failure.

I noticed that without global cache, the name of source folder is always `foobar-src`, no version or some tag information entangled, which prevented re-downloading of the dependency; and the old folder is shallow-cloned, therefore resulted in checkout failure.

So I guess your suggestion is necessary to keep the local cache work.

Btw: the logging output is deleted.